mypy/nodes.py

```diff
@@ -3017,6 +3017,24 @@ class is generic then it will be a type constructor of higher kind.
         "is_intersection",
     ]
 
+    # Special attributes not collected as protocol members by Python 3.12
+    # See typing._SPECIAL_NAMES
+    EXCLUDED_ATTRIBUTES: Final = frozenset(
+        {
+            "__abstractmethods__",
+            "__annotations__",
+            "__dict__",
+            "__doc__",
+            "__init__",
+            "__module__",
+            "__new__",
```

Comment:
I am also not sure about `__new__`. I think that there might be valid use-case for it.